src/transaction/txParser.ts

```diff
@@ -183,34 +225,58 @@ const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
   )
 }
 
-const parseTxWithdrawals = (
-  withdrawals: any,
-): _Withdrawal[] => {
+const parseTxWithdrawals = (withdrawals: any): _Withdrawal[] => {
   if (!isWithdrawalsMap(withdrawals)) {
     throw Error(Errors.WithrawalsParseError)
   }
   return Array.from(withdrawals).map(([address, coins]): _Withdrawal => ({ address, coins: BigInt(coins) }))
 }
 
-const parseTtl = (ttl: any): number => {
-  if (ttl === null || ttl === undefined) {
-    throw Error(Errors.MssingTTLParseError)
+const parseFee = (fee: any): Lovelace => {
+  if (!isUint64(fee)) {
+    throw Error(Errors.FeeParseError)
+  }
+  return BigInt(fee)
+}
+
+const parseTtl = (ttl: any): BigInt | undefined => {
```

Comment:
optional return types should rather be `type | null` - undefined hampers overall "inspectability" of the code as everything is undefined by default whereas null positively proves that the value has been set at some point. Or is there some strong reason to stick to undefined?

Reply:
No strong reason, I just thought it's more clear to have types defined as
```
ttl?: BigInt,
```
compared to
```
ttl: BigInt | null,
```